Mixpanel/MPNotificationViewController.m

```diff
@@ -215,44 +234,39 @@ - (void)viewDidLoad
     _canPan = YES;
     _isBeingDismissed = NO;
     self.view.clipsToBounds = YES;
+    
+    MPMiniNotification *notification = (MPMiniNotification *) self.notification;
 
     self.imageView = [[UIImageView alloc] initWithFrame:CGRectZero];
     self.imageView.layer.masksToBounds = YES;
 
     self.bodyLabel = [[UILabel alloc] initWithFrame:CGRectZero];
-    self.bodyLabel.textColor = [UIColor whiteColor];
+    self.bodyLabel.textColor = UIColorFromRGB(notification.bodyColor);
     self.bodyLabel.backgroundColor = [UIColor clearColor];
     self.bodyLabel.font = [UIFont systemFontOfSize:14.0f];
     self.bodyLabel.lineBreakMode = NSLineBreakByWordWrapping;
     self.bodyLabel.numberOfLines = 0;
 
-    [self initializeMiniNotification];
+    self.view.backgroundColor = UIColorFromRGB(notification.backgroundColor);
 
     if (self.notification != nil) {
         if (self.notification.image != nil) {
-            self.imageView.image = [UIImage imageWithData:self.notification.image scale:2.0f];
+            self.imageView.image = [UIImage imageWithData:notification.image scale:2.0f];
+            UIImage *originalImage = [UIImage imageWithData:notification.image scale:2.0f];
+            UIImage *tintedImage = [originalImage imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
+            [self.imageView setImage:tintedImage];
+            self.imageView.tintColor = UIColorFromRGB(notification.imageTintColor);
             self.imageView.hidden = NO;
         } else {
             self.imageView.hidden = YES;
         }
 
         if (self.notification.body != nil) {
-            self.bodyLabel.text = self.notification.body;
+            self.bodyLabel.text = notification.body;
```

Comment:
should we also change all the checks to `notification.x != nil` since we changed what we're setting from `self.notification.body` to `notification.body`? happens in other parts of this file too